### DISCUSSION TITLE
How to Insert HTML into dangerouslySetInnerHTML?

### DISCUSSION THREAD

**Discussion by Uranbold** [vercel/next.js#13143] [Help]
Hello,

We got strange question.

We had Rest API and that gives us HTML content but somehow we need to insert 1 div into the Rest API content.

Example: 
```
<div>
<div>Come from API 1</div>
<div>Come from API 2</div>
</div>
```

```
<div>
<div>Come from API 1</div>
<div>Our DIV from React Component</div>
<div>Come from API 2</div>
</div>
```

Thanks.

**Comment by brunocrosier** [comment]
could you do something like 

```javascript
const foo = encodeURI(`
<div>
<div>Come from API 1</div>
<div>Come from API 2</div>
</div>
`
```

and then just use `decodeURI(foo)` afterwards ?

**Reply by Uranbold** [reply]
Hello,

I mean this .content is come from the Rest API content field.
```
<div class="content">
<div>Sample content</div>
<div>Sample content 2</div>
</div>
```
I am trying to insert it into that content to my own component div.

Thanks.

**Comment by brunocrosier** [comment]
Something like this?

```javascript


<div className="wrapper"
dangerouslySetInnerHTML={{
          __html: `${contentFromYourAPI}`
        }}
/>


```

**Reply by Uranbold** [reply]
Output should something like this ->

<div class="content">
<div>a</div>
<div>c</div>
<div>b</div>
</div>
```
That C should inserted from my Component :)

Thanks.

**Comment by brunocrosier** [comment]
Yes, I understand. The answer to your question is this:

```javascript
<div className="wrapper"
dangerouslySetInnerHTML={{
          __html: `${contentFromYourAPI}`
        }}
/>
```

just replace `${contentFromYourAPI}` with your HTML and it will work
